src/variables/variable.jl

```diff
@@ -93,13 +93,29 @@ getmarginals(variables::AbstractArray{<:AbstractVariable}, skip_strategy::Margin
 
 ### Marginals
 
+struct RepeatedValueContainer
+    marginal
+    container
+    _length
+end
+
+Base.length(ri::RepeatedValueContainer) = ri._length
+Base.iterate(ri::RepeatedValueContainer) = iterate(ri.container)
+Base.iterate(ri::RepeatedValueContainer, state) = iterate(ri.container, state)
+
+function RepeatedValueContainer(marginal, _length)
+    container = Iterators.repeated(marginal, _length)
+    return RepeatedValueContainer(marginal, container, _length)
+end
+
```

Comment:
```suggestion
```
Why is this needed? (+ it is slower than just using `Iterators.repeated` because all types are `Any`)

Reply:
@bvdmitri you are right It's not needed, my reasoning about iterators length function was wrong. I checked the docs.